pretrain_gpt_alcf.py

```diff
@@ -498,13 +512,16 @@ def train_valid_test_datasets_provider(train_val_test_num_samples):
         '> building train, validation, and test datasets for GPT ...'
     )
     files = []
+    # making it the default input method
+    assert(args.data_file_list is not None)
     if args.data_file_list is not None:
         log.info(f"Reading datasets from {args.data_file_list}")
         with open(args.data_file_list, 'r') as flist:
             for f in flist.readlines():
-                w, fname = f.split()
+                w, fname, c = f.split()
```

Comment:
@zhenghh04 just to make sure I understand:

- this change ^, i.e.

    ```python
    w, fname, c = f.split()
    ```

    will incorporate the "corpus" that each sample is coming from, right?

    so, each line in our `DATA_FILE_LIST` should be of the form:

    ```bash
    # weight /path/to/some_text_document corpus
    # e.g.:
    0.003 /path/to/books-0000_text_document books
    0.0002 /path/to/arxiv-0000_text_document arxiv
    ```
  
  correct?

Reply:
(I think so, since it worked with a new `DATA_FILE_LIST` following this scheme)

In this case, as I was testing / working on aggregating across different corpora and (accidentally) tried running with a `args.data_file_list` that contained a blankline.

e.g.

```bash
0.003 /path/to/books-0000_text_document books

0.0002 /path/to/arxiv-0000_text_document arxiv
```

will crash with  

`ValueError: not enough values to unpack (expected 3, got 0)` 

(coming from the blank line)

<details closed><summary><code>Traceback</code>:</summary>

```bash
Traceback (most recent call last):
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/pretrain_gpt_alcf.py", line 646, in <module>
    model = main()
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/pretrain_gpt_alcf.py", line 605, in main
    model = pretrain(
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/megatron/training.py", line 246, in pretrain
    = build_train_valid_test_data_iterators(
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/megatron/training.py", line 1723, in build_train_valid_test_data_iterators
    build_train_valid_test_data_loaders(
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/megatron/training.py", line 1679, in build_train_valid_test_data_loaders
    train_ds, valid_ds, test_ds = build_train_valid_test_datasets(
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/megatron/training.py", line 1651, in build_train_valid_test_datasets
    return build_train_valid_test_datasets_provider(train_val_test_num_samples)
  File "/lus/gecko/projects/Aurora_deployment/foremans/projects/argonne-lcf/Megatron-DeepSpeed/pretrain_gpt_alcf.py", line 521, in train_valid_test_datasets_provider
    w, fname, c = f.split()
ValueError: not enough values to unpack (expected 3, got 0)
```

</details>

maybe not a huge deal, but I guess we could fix with something like ??


```python
with open(args.data_file_list, 'r') as flist:
    for f in flist.readlines():
        if len(line.strip()) != 0:
            w, fname, c = f.split()
```

🤷🏻‍♂️

Reply:
fixed in [bf979a74ae72d62a1d77617273b5a6b9a07ba5f7](https://github.com/argonne-lcf/Megatron-DeepSpeed/pull/16/commits/bf979a74ae72d62a1d77617273b5a6b9a07ba5f7)